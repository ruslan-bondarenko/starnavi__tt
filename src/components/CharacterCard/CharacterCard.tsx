import React, { FC } from "react";
import Link from "next/link";

import { ICharacter } from "@/shared/types";
import { getIdFromUrl } from "@/shared";

type Props = {
  character: ICharacter;
};

const ProductItem: FC<Props> = ({ character }) => {
  return (
    <Link href={`/people/${getIdFromUrl(character?.url || "")}`}>
      <h4 className="px-4 py-3 text-l font-medium truncate">
        {character?.name}
      </h4>
    </Link>
  );
};

export default ProductItem;
