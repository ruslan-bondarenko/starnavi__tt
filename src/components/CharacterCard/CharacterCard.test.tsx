import { render, screen } from "@testing-library/react";

import CharacterCard from "./CharacterCard";
import { describe, it } from "@jest/globals";

declare const expect: jest.Expect;

const character = {
  name: "Obi-Wan Kenobi",
  height: "182",
  mass: "77",
  hair_color: "auburn, white",
  skin_color: "fair",
  eye_color: "blue-gray",
  birth_year: "57BBY",
  gender: "male",
  homeworld: 20,
  films: [1, 2, 3, 4, 5, 6],
  species: [1],
  vehicles: [38],
  starships: [48, 59, 64, 65, 74],
  created: "2014-12-10T16:16:29.192000Z",
  edited: "2014-12-20T21:17:50.325000Z",
  url: "https://sw-api.starnavi.io/people/10/",
};

describe("CharacterCard component", () => {
  it("CharacterCard renders", () => {
    render(<CharacterCard character={character} />);

    expect(screen.getByText("Obi-Wan Kenobi")).toBeInTheDocument();
  });

  it("CharacterCard snapshot", () => {
    const card = render(<CharacterCard character={character} />);

    expect(card).toMatchSnapshot();
  });
});
