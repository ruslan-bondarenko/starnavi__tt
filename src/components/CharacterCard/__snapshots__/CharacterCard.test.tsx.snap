// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard component CharacterCard snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="/people/10"
      >
        <h4
          class="px-4 py-3 text-l font-medium truncate"
        >
          Obi-Wan Kenobi
        </h4>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="/people/10"
    >
      <h4
        class="px-4 py-3 text-l font-medium truncate"
      >
        Obi-Wan Kenobi
      </h4>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
